index/evaluation.py

```diff
@@ -177,3 +271,69 @@ def score_mappings(matches: pd.DataFrame) -> float:
     matches = matches[matches["target_concept_label"].notnull()]
     accuracy = matches["correct"].sum() / len(matches)
     return accuracy
+
+
+def evaluate(
+    datasets,
+    labels,
+    store_results=False,
+    model="gpt",
+    results_root_dir="resources/results/pd",
+):
+
+    if model == "gpt":
+        data_gpt = {}
+        data_fuzzy = {}
+        for idx, source in enumerate(datasets):
+            acc_gpt = []
+            acc_fuzzy = []
+            for idy, target in enumerate(datasets):
+                map_gpt = match_closest_descriptions(source, target)
+                map_fuzzy = match_closest_descriptions(
+                    source, target, matching_method=MatchingMethod.FUZZY_STRING_MATCHING
+                )
+                if target == "jadni":
+                    print("check")
+                if store_results:
+                    map_gpt.to_excel(
+                        results_root_dir
+                        + "/gpt_"
+                        + f"{labels[idx]}_to_{labels[idy]}.xlsx"
+                    )
+                    map_fuzzy.to_excel(
+                        results_root_dir
+                        + "/fuzzy_"
+                        + f"{labels[idx]}_to_{labels[idy]}.xlsx"
+                    )
+                acc_gpt.append(round(score_mappings(map_gpt), 2))
+                acc_fuzzy.append(round(score_mappings(map_fuzzy), 2))
+            data_gpt[labels[idx]] = acc_gpt
+            data_fuzzy[labels[idx]] = acc_fuzzy
+        # transpose to have from -> to | row -> column like in the paper
+        gpt = pd.DataFrame(data_gpt, index=labels).T
+        fuzzy = pd.DataFrame(data_fuzzy, index=labels).T
+        return gpt, fuzzy
+
+    elif model == "mpnet":
+        data_mpnet = {}
+        for idx, source in enumerate(datasets):
+            acc_mpnet = []
+            for idy, target in enumerate(datasets):
+                map_mpnet = match_closest_descriptions(
+                    source,
+                    target,
+                    matching_method=MatchingMethod.COSINE_EMBEDDING_DISTANCE,
+                )
+                if target == "jadni":
+                    print("check")
+                if store_results:
+                    map_mpnet.to_excel(
+                        results_root_dir
+                        + "/mpnet_"
+                        + f"{labels[idx]}_to_{labels[idy]}.xlsx"
+                    )
+                acc_mpnet.append(round(score_mappings(map_mpnet), 2))
+            data_mpnet[labels[idx]] = acc_mpnet
+        # transpose to have from -> to | row -> column like in the paper
+        mpnet = pd.DataFrame(data_mpnet, index=labels).T
+        return mpnet
```

Comment:
Having a function return a different number of values for different arguments is not a good idea, since it makes the behaviour somewhat less predictable to the user.

Better solution would be here to have it return either gpt/fuzzy/mpnet results (since you already have a model paprameter) .